core/src/test/java/org/apache/iceberg/FileGenerationUtil.java

```diff
@@ -0,0 +1,187 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+import org.apache.iceberg.io.DeleteSchemaUtil;
+import org.apache.iceberg.io.LocationProvider;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Types;
+
+public class FileGenerationUtil {
+
+  private static final Random RANDOM = new Random();
+
+  private FileGenerationUtil() {}
+
+  public static DataFile generateDataFile(Table table, StructLike partition) {
+    Schema schema = table.schema();
+    PartitionSpec spec = table.spec();
+    LocationProvider locations = table.locationProvider();
+    String path = locations.newDataLocation(spec, partition, generateFileName());
+    long fileSize = generateFileSize();
+    Metrics metrics = generateRandomMetrics(schema);
+    return DataFiles.builder(spec)
+        .withPath(path)
+        .withPartition(partition)
+        .withFileSizeInBytes(fileSize)
+        .withFormat(FileFormat.PARQUET)
+        .withMetrics(metrics)
+        .build();
+  }
+
+  public static DeleteFile generatePositionDeleteFile(Table table, StructLike partition) {
+    PartitionSpec spec = table.spec();
+    LocationProvider locations = table.locationProvider();
+    String path = locations.newDataLocation(spec, partition, generateFileName());
+    long fileSize = generateFileSize();
+    Metrics metrics = generatePositionDeleteMetrics();
+    return FileMetadata.deleteFileBuilder(table.spec())
+        .ofPositionDeletes()
+        .withPath(path)
+        .withPartition(partition)
+        .withFileSizeInBytes(fileSize)
+        .withFormat(FileFormat.PARQUET)
+        .withMetrics(metrics)
+        .build();
+  }
+
+  public static DeleteFile generatePositionDeleteFile(Table table, DataFile dataFile) {
+    PartitionSpec spec = table.spec();
+    StructLike partition = dataFile.partition();
+    LocationProvider locations = table.locationProvider();
+    String path = locations.newDataLocation(spec, partition, generateFileName());
+    long fileSize = generateFileSize();
+    Metrics metrics = generatePositionDeleteMetrics(dataFile);
+    return FileMetadata.deleteFileBuilder(table.spec())
+        .ofPositionDeletes()
+        .withPath(path)
+        .withPartition(partition)
+        .withFileSizeInBytes(fileSize)
+        .withFormat(FileFormat.PARQUET)
+        .withMetrics(metrics)
+        .build();
+  }
+
+  public static String generateFileName() {
+    int partitionId = RANDOM.nextInt(100_000);
```

Comment:
Do we need a partition id in the file name? Files will locate in the partition dir anyway.